Packages/io.chainsafe.web3-unity.loader/Editor/ScopedRegistryInstaller.cs

```diff
@@ -0,0 +1,125 @@
+#if UNITY_EDITOR
+using UnityEditor;
+using UnityEngine;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Newtonsoft.Json.Linq;
+
+[InitializeOnLoad]
+public static class ScopedRegistryAndDependencyInstaller
+{
+    private static readonly string RegistryName = "package.openupm.com";
+    private static readonly string RegistryUrl = "https://package.openupm.com";
+    private static readonly string[] RequiredScopes =
+    {
+        "com.reown.appkit.unity",
+        "com.nethereum.unity",
+        "com.reown.core",
+        "com.reown.core.common",
+        "com.reown.core.crypto",
+        "com.reown.core.network",
+        "com.reown.core.storage",
+        "com.reown.sign",
+        "com.reown.sign.nethereum",
+        "com.reown.sign.nethereum.unity",
+        "com.reown.sign.unity",
+        "com.reown.unity.dependencies"
+    };
+
+    // The Git dependency to add
+    private const string ChainsafeDependencyKey = "io.chainsafe.web3-unity";
+    private const string ChainsafeLoaderDependencyKey = "io.chainsafe.web3-unity.loader";
+    private const string ChainsafeDependencyUrl = "https://github.com/ChainSafe/web3.unity.git?path=/Packages/io.chainsafe.web3-unity#nikola/appkit-implementation-1210";
+    private const string DependenciesKey = "Dependencies Installed";
+    static ScopedRegistryAndDependencyInstaller()
+    {
+        // Check if we've already installed the registry and dependencies
+        if (PlayerPrefs.GetInt(DependenciesKey, 0) == 1)
```

Comment:
you can editor prefs instead here, we should also use the `EditorApplication.projectChanged` which has two advantages
1. event invokes even with compile errors
2. event doesn't invoke every time we recompile